tephra-core/src/main/java/co/cask/tephra/distributed/ElasticPool.java

```diff
@@ -148,38 +160,10 @@ public void release(T element) {
    * @param element
    */
   public void discard(T element) {
```

Comment:
When will this method gets called? In general, it should be the pool who decides when to discard an element, not through from external party.